### ISSUE TITLE
Trying to implement without CordovaActivity

### ISSUE THREAD

**Issue by JonnyBGod** [axemclion/react-native-cordova-plugin#10]
I am trying to put this to work without extending CordovaActivity. I need to do this because I am using another library for which I need to extend FragmentActivity.

At the moment I am getting the following errors:

```
MainActivity.java:108: error: cordovaPluginAdapter has protected access in CordovaPluginPackage
        CordovaInterfaceImpl cordovaInterface = cordovaPluginPackage.cordovaPluginAdapter.cordovaInterface;
                                                                    ^
MainActivity.java:108: error: cordovaInterface has protected access in CordovaPluginAdapter
        CordovaInterfaceImpl cordovaInterface = cordovaPluginPackage.cordovaPluginAdapter.cordovaInterface;
```

My code:

```
package com.realityConnect.mckrides;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.KeyEvent;

import com.facebook.react.LifecycleState;
import com.facebook.react.ReactInstanceManager;
import com.facebook.react.ReactRootView;
import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;

import android.support.v4.app.FragmentActivity;
import com.aakashns.reactnativedialogs.ReactNativeDialogsPackage;

import io.cordova.reactnative.CordovaPluginPackage;
import org.apache.cordova.CordovaInterfaceImpl;
import org.apache.cordova.LOG;

public class MainActivity extends FragmentActivity implements DefaultHardwareBackBtnHandler {

    private ReactInstanceManager mReactInstanceManager;
    private ReactRootView mReactRootView;
    private static final String TAG = "CORDOVA_ACTIVITY";
    protected CordovaPluginPackage cordovaPluginPackage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mReactRootView = new ReactRootView(this);

        mReactInstanceManager = ReactInstanceManager.builder()
                .setApplication(getApplication())
                .setBundleAssetName("index.android.bundle")
                .setJSMainModuleName("index.android")
                .addPackage(new MainReactPackage())

                .addPackage(new CordovaPluginPackage(this, savedInstanceState))
                .addPackage(new ReactNativeDialogsPackage(this))

                .setUseDeveloperSupport(true)
                .setInitialLifecycleState(LifecycleState.RESUMED)
                .build();

        mReactRootView.startReactApplication(mReactInstanceManager, "McRides", null);

        setContentView(mReactRootView);
    }

...

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
        CordovaInterfaceImpl cordovaInterface = cordovaPluginPackage.cordovaPluginAdapter.cordovaInterface;
        LOG.d(TAG, "Incoming Result. Request code = " + requestCode);
        super.onActivityResult(requestCode, resultCode, intent);
        cordovaInterface.onActivityResult(requestCode, resultCode, intent);
    }
}

```

**Comment by axemclion** [comment]
@JonnyBGod Looks like I need to make cordovaPluginAdapter public. Will change this and publish a new version.

**Comment by axemclion** [comment]
@JonnyBGod Pushed the new 0.0.8 version to npm, so that this issue is fixed. 

Btw, I am working on moving the entire toolchain to start using [rnpm](https://github.com/rnpm/rnpm) - that will make it much more clean, and all the manual steps will be removed

**Comment by ncnlinh** [comment]
@axemclion the fix did not help for me. I suppose the changes should be made in `CordovaPluginPackage` and `CordovaPluginAdapter`?

**Comment by axemclion** [comment]
@ncnlinh I just published 1.0.0 version of this plugin on npm. I also now have an [example app](https://github.com/axemclion/react-native-cordova-plugin/tree/examples) that shows how to use the plugins.
